pkg/k8s/version/version.go

```diff
@@ -49,6 +48,10 @@ type ServerCapabilities struct {
 
 	// EndpointSlice is the ability of k8s server to support endpoint slices
 	EndpointSlice bool
+
+	// LeasesResourceLock is the ability of K8S server to support Lease type
+	// from CoordinationV1 API for leader election purposes.
```

Comment:
Nit: 

```suggestion
	// LeasesResourceLock is the ability of K8s server to support Lease type
	// from CoordinationV1 API for leader election purposes.
```

Could you also add a reference to the K8s docs in this comment? I've [mentioned previously in another PR](https://github.com/cilium/cilium/pull/12611#pullrequestreview-455054054) that we should more thoroughly document when we make changes to the K8s version / capabilities related code.

At a high-level it would be nice to have:

- A reference link to K8s docs so that our future selves can understand what this thing is and potentially why we need it.
- The relevant versions this capability maps to, so that when we remove / add support for certain K8s versions, we are able to know if this applicability is relevant or not.

Reply:
To add another reason why I think it's important to document these things, is because we rarely make changes to this code / package[1]. It averages to around once every ~3 months, so lots of time / opportunity to lose context.

[1]: `git log --oneline --format=format:"%cs %s" pkg/k8s/version`

Reply:
Sounds good, I will refactor the documentation.